browser/ui/webui/settings/brave_default_extensions_handler.cc

```diff
@@ -112,6 +166,28 @@ void BraveDefaultExtensionsHandler::OnInstallResult(
   }
 }
 
+void BraveDefaultExtensionsHandler::SetMediaRouterEnabled(
+    const base::ListValue* args) {
+  CHECK_EQ(args->GetSize(), 1U);
+  CHECK(profile_);
+  bool enabled;
+  args->GetBoolean(0, &enabled);
+
+  restart_needed_ = !restart_needed_;
+  // Adding the media router extension on explicit toggle to prevent
+  // any suprise connections on upgrade
+  if (enabled) {
+    extensions::ExtensionPrefs::Get(profile_)->SetExtensionEnabled(
```

Comment:
we shouldn't be directly calling `ExtensionPrefs::SetExtensionEnabled`. You should enable/disable using the `ExtensionRegistry`